test/object-store/sync/app.cpp

```diff
@@ -2021,23 +2046,177 @@ TEST_CASE("app: sync integration", "[sync][app]") {
 
     class HookedTransport : public SynchronousTestTransport {
     public:
-        void send_request_to_server(Request&& request,
-                                    util::UniqueFunction<void(const Response&)>&& completion_block) override
+        void send_request_to_server(Request&& request, HttpCompletion&& completion_block) override
         {
             if (request_hook) {
                 request_hook(request);
             }
-            SynchronousTestTransport::send_request_to_server(std::move(request), [&](const Response& response) {
-                if (response_hook) {
-                    response_hook(request, const_cast<Response&>(response));
-                }
-                completion_block(response);
-            });
+            if (simulated_response) {
+                return completion_block(std::move(request), *simulated_response);
+            }
+            SynchronousTestTransport::send_request_to_server(
+                std::move(request), [&](const Request& request, const Response& response) {
+                    if (response_hook) {
+                        response_hook(const_cast<Request&>(request), const_cast<Response&>(response));
+                    }
+                    completion_block(std::move(request), std::move(response));
+                });
         }
+        // Optional handler for the request and response before it is returned to completion
         util::UniqueFunction<void(Request&, Response&)> response_hook;
+        // Optional handler for the request before it is sent to the server
         util::UniqueFunction<void(Request&)> request_hook;
+        // Optional Response object to return immediately instead of communicating with the server
+        util::Optional<Response> simulated_response;
     };
 
+    {
+        std::unique_ptr<realm::AppSession> app_session;
+        std::string base_file_path = util::make_temp_dir() + random_string(10);
+        auto redir_transport = std::make_shared<HookedTransport>();
+        AutoVerifiedEmailCredentials creds;
+
+        auto app_config = get_config(redir_transport, session.app_session());
+        set_app_config_defaults(app_config, redir_transport);
+
+        util::try_make_dir(base_file_path);
+        SyncClientConfig sc_config;
+        sc_config.base_file_path = base_file_path;
+        sc_config.log_level = TEST_ENABLE_SYNC_LOGGING ? util::Logger::Level::all : util::Logger::Level::off;
```

Comment:
You can just use this define https://github.com/realm/realm-core/blob/master/test/object-store/util/test_file.hpp#L228 to setup the log level if you've included test_file.hpp already.